### DISCUSSION TITLE
Binary caching webdav or S3 backend support

### DISCUSSION THREAD

**Discussion by xiaozhuai** [microsoft/vcpkg#25039] [Ideas]
It would be awesome if vcpkg could support storing binary cache on a webdav or S3 server.
I considered to mount webdav or S3 to filesystem. But there would be many problems.
It will be better if vcpkg support webdav or S3 directly.

**Comment by JackBoosY** [comment]
Yeah, although vcpkg provides binarycaching functionality, we still do not currently provide precompiled binaries.

**Reply by xiaozhuai** [reply]
Precompiled binaries would be a good option. I just want to share binary caching through my team members, so it would be better if we can set a webdav or S3 url for caching.

**Comment by autoantwort** [comment]
Maybe you can use the x-aws binary source
<img width="677" alt="image" src="https://user-images.githubusercontent.com/41973254/171684636-373ac9f3-a632-4249-8c9d-c5ae76766cca.png">

**Comment by xiaozhuai** [comment]
@autoantwort Tried this with no luck. I use [MinIO](https://docs.min.io/docs/aws-cli-with-minio.html). There is no way that I can set an endpoint url.

**Reply by contagon** [reply]
Just in case anyone else lands here, this is possible to do via the [AWS_ENDPOINT_URL](https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-endpoints.html) environment vairable in aws-cli version > 2.13.0+.
